### ISSUE TITLE
Mayavi Source for Simphony Engine

### ISSUE THREAD

**Issue by kitchoi** [simphony/simphony-mayavi#109]
I added a `EngineSource` (similar to `CUDSFileSource`) for handling datasets from a simphony modeling engine.  Then I added a `EngineSourceManager` for visualisation and basic operations of the engine.  The screenshot below is the `EngineSourceManager` in action.  (I actually used it on an engine from `simphony-lammps`)
![engine_source_manager](https://cloud.githubusercontent.com/assets/3673984/11981733/25d1f134-a99d-11e5-93b3-765ba98abbf7.png)

@dpinte, @itziakos, the new feature is currently available in the branch `add-engine-source`.  I still need to add more tests and documentation before I can make a PR.  At the moment I would like to solicit feedbacks.

If you could check out the branch, there is a dummy engine so you can experiment with the following:

```
>>> from simphony_mayavi.sources.tests import testing_utils
>>> engine = testing_utils.DummyEngine()
>>> from simphony_mayavi.plugins.api import EngineSourceManager
>>> manager = EngineSourceManager(engine)
>>> manager.show_config()   # show the UI
```

If you dig into the code you will find some in-function imports.  That's because I intend to make it available as a plugin in Mayavi ([ref](http://docs.enthought.com/mayavi/mayavi/auto/example_user_mayavi.html))

Please let me know if you have any comments/suggestions.

**Comment by kitchoi** [comment]
I have updated the GUI.  To test it, check out the add-engine-source branch and try this:

```
from simphony_mayavi.plugins.api import EngineManagerStandaloneUI
from simphony_mayavi.sources.tests import testing_utils
engine = testing_utils.DummyEngine()
manager = EngineManagerStandaloneUI("test", engine)
manager.show_config()
```

![action1-new](https://cloud.githubusercontent.com/assets/3673984/12067974/af75eed6-affb-11e5-9699-29cf3ce1ce4b.png)
![action2-new](https://cloud.githubusercontent.com/assets/3673984/12067976/b3714e9a-affb-11e5-988b-3fdfed26389d.png)

**Comment by kitchoi** [comment]
I have added a mayavi2 plugin so that a panel view can be added in Mayavi2 and the user can add engine via the python shell in Mayavi.  However, this plugin requires envisage to be included in the setup requirement.
![mayavi2-plugin](https://cloud.githubusercontent.com/assets/3673984/12151013/c4d567c2-b4a4-11e5-96ca-2508a61995a3.png)

**Comment by kitchoi** [comment]
merged #110
